src/plugins/analysis/plugin.py

```diff
@@ -0,0 +1,393 @@
+from __future__ import annotations
+
+import abc
+import ctypes
+import io
+import logging
+import queue
+import re
+import signal
+import time
+import typing
+from multiprocessing import Array, Process, Queue, Value
+
+from pydantic import validator
+from pydantic.dataclasses import dataclass
+
+from objects.file import FileObject
+from storage.fsorganizer import FSOrganizer
+
+# TODO should this be configurable?
+ANALYSIS_STATS_LIMIT = 1000
```

Comment:
At the time no plugin set this value to anything other than the default. Is there any reason to have it configurable?